### DISCUSSION TITLE
Why AsMaybe for structs has `ref`?

### DISCUSSION THREAD

**Discussion by marcinjahn** [vkhorikov/CSharpFunctionalExtensions#575] [General]
Have a look at https://github.com/vkhorikov/CSharpFunctionalExtensions/blob/master/CSharpFunctionalExtensions/Maybe/Extensions/AsMaybe.cs#L16.

I guess the `ref` is there to avoid copying of the argument. However, it makes it difficult to use the extensions with properties.

Example:

```cs
public record MyRecord(int? SomeInt);

var instance = new MyRecord(3);

var maybe = instance.SomeInt.AsMaybe(); // errors with "a non ref-returning property or indexer may not be used as an out or ref value"
```

the workaround is to assign to some variable first:

```cs
var someInt = instance.SomeInt;
var maybe = instance.SomeInt.AsMaybe();
```

Is there some way to make the experience a bit better?

**Comment by vkhorikov** [comment]
I guess we can get rid of `ref` there. The difference shouldn't be noticeable. Feel free to submit a PR (if this solution works; I haven't tested it)
